generator/src/main/java/com/scottlogic/deg/generator/restrictions/DateTimeRestrictionsMerger.java

```diff
@@ -43,17 +78,35 @@
             return left;
         }
 
+        // if both exclusive
+        if (!(left.isInclusive() && right.isInclusive())){
+
+            // if both datetimes sre the same when granularity applied
+            if(granularity.getGranularityFunction().apply(left.getLimit()).equals(granularity.getGranularityFunction().apply(right.getLimit()))){
+                inclusiveOverride = true;
+            }
+        }
+
+        // if left and right are identical, return new object with same values
         if (left.getLimit().compareTo(right.getLimit()) == 0)
             return new DateTimeRestrictions.DateTimeLimit(left.getLimit(), left.isInclusive() && right.isInclusive());
-        switch(mergeLimit) {
+
+        switch (mergeLimit) {
             case MIN:
-                if (left.getLimit().compareTo(right.getLimit()) > 0)
-                    return left;
-                return right;
+                if (left.getLimit().compareTo(right.getLimit()) > 0){
+                    return new DateTimeRestrictions.DateTimeLimit(left.getLimit(), (left.isInclusive() || right.isInclusive()) || inclusiveOverride);
```

Comment:
`left.isInclusive() || right.isInclusive()) || inclusiveOverride` is repeated at least 3 times, can we make this a well named variable instead?